src/widgets/training-program/ui/training-program.tsx

```diff
@@ -1,26 +1,30 @@
 import { cloneElement } from 'react';
 import classNames from 'classnames/bind';
 import Image from 'next/image';
-import type { Course } from '@/entities/course';
-import { Language } from '@/shared/types';
+import { isTrainingProgramType } from '@/shared/helpers/is-training-program';
+import { selectCourse } from '@/shared/hooks/use-course-by-title/utils/select-course';
 import { LinkCustom } from '@/shared/ui/link-custom';
 import { WidgetTitle } from '@/shared/ui/widget-title';
-import { TrainingProgramType, contentMap, trainingProgramLink } from 'data';
+import { CourseNamesKeys, contentMap, trainingProgramLink } from 'data';
 
 import styles from './training-program.module.scss';
 
 const cx = classNames.bind(styles);
 
 type TrainingProgramProps = {
-  courseName: TrainingProgramType;
-  lang?: Language;
-  course: Course;
+  courseName: CourseNamesKeys;
+  specify?: string;
 };
 
-export const TrainingProgram = ({ courseName, lang = 'en', course }: TrainingProgramProps) => {
-  const { title, content, image } = contentMap[courseName];
+export const TrainingProgram = async ({ courseName, specify = '' }: TrainingProgramProps) => {
+  const course = await selectCourse(courseName);
+  const { language } = course;
+  const programName = `${courseName} ${specify}`;
+  let contentName = isTrainingProgramType(programName) ? programName : courseName;
```

Comment:
I find the logic behind this very confusing
Correct me if I'm wrong, but it returns `false` for the first case just because it has an extra space
![image](https://github.com/user-attachments/assets/e1f60215-1e3c-4e6a-809c-f57a80d35de7)
![image](https://github.com/user-attachments/assets/f010546a-d8ac-461a-a763-3c33e452b8c8)

Reply:
We have issue for separate badge component

Reply:
ref: https://github.com/rolling-scopes/site/issues/589